src/Components/Components/src/ComponentRouteData.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Describes information determined during routing that specifies
+    /// the page to be displayed.
+    /// </summary>
+    public sealed class ComponentRouteData
```

Comment:
I have nits to pick with all of these names... could we drop the prefix `Component`? We already have duplicate names for `[Route]`

Reply:
Do you really think it's OK to clash with ASP.NET Core's existing `RouteData` type? I take your point that we do for `RouteAttribute`, but adding more clashes doesn't seem like an improvement.

I'm concerned that this is not only confusing for people writing Blazor apps, but also any other ASP.NET Core app since it's all shared framework. Once you type RouteData` and VS offers to add either namespace, there's plenty of potential for confusion.

If you feel strongly that `RouteData` is the better name, I won't object to changing it.

Reply:
I feel pretty strongly about not having `Component` in the name. Using framework-level terms in prefixes, just makes stuff feel not-first-class. I'm totally happy to duplicate the name, or pick something else that doesn't duplicate but is a better name overall.

Other naming ideas?

Reply:
Renamed to `RouteData`